### PR TITLE
Fix z-indexing on /documentation

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Fix z-indexing on /documentation
- Make navigation drawer flush with navigation

## QA

- Go to https://canonical-com-804.demos.haus/documentation on mobile view or a small screen
- Check that when a drop down from the top-navigation is open, it goes over the side-navigation
- Check the navigation drawer is flush with the navigation.

## Issue / Card

Fixes https://github.com/canonical/canonical.com/issues/803

## Screenshots

before:
image
after:
image

before:
image
after:
image

before:

### CODE REVIEW

File: templates/documentation/beyond-canonical.html [hunk 1]
```diff
@@ -14,12 +14,12 @@
 <div class="p-strip u-no-padding--top">
   <div class="row" id="documentation">
     <aside class="col-3">
-      <div class="p-side-navigation--raw-html is-sticky" id="drawer" style="top: 4.5rem; z-index: 37;">
+      <div class="p-side-navigation--raw-html is-sticky" id="drawer">
```
Comment (carkod) [inline templates/documentation/beyond-canonical.html:17]: Not entirely sure if this is the reason, but now the first element of the navigation is "overscrolling".


https://user-images.githubusercontent.com/14939793/220315502-63f60f25-b170-4161-999e-98048a119bcb.mov

Comment (petesfrench) [reply templates/documentation/beyond-canonical.html:17]: Good catch. I have updated my PR, could you take another look please

**Comment by carkod** [comment]
The choice of white for current navigation looks a bit weird on this page, it looks like it's leaking, but it's not a discussion for this PR, just thinking out loud
Screenshot 2023-02-21 at 15 56 27

**Review by carkod — APPROVED** [review]

**Comment by petesfrench** [comment]
@carkod I think this should mostly be alleviated with the new surus, as they are grey, or off white in color

**Pull request merged**